### DISCUSSION TITLE
Fork with a few ideas

### DISCUSSION THREAD

**Discussion by regbo** [dizitart/jbus#24] [Ideas]
Hey, great project. I created a fork here with a few changes:
https://github.com/regbo/jbus

The changes include:
- token replacement in logging (if an event has an expensive toString() that can slow logging down)
- removed a lot of the error logging, as I figure that should just be thrown
- custom thread pools, for fine grained tuning
- added the ability to throw an exception on no listeners
- made listeners listen to subclasses
- add type arguments to jbus and a special Listener class. if you use JBus<Object> it's the same as the master, but if you use JBus<Date> you can add lambdas in java 8 like so:
```
		JBus<Date> jbus = new JBus<>(Date.class, Executors.newCachedThreadPool());
		jbus.register(d -> {
			System.out.println("the date is:" + d);
		});
		jbus.post(new Date());
```
- added thread tracking so that static deregister calls can be made from within listeners. for example, the following will only print once, not 3 times:
```
		JBus<Date> jbus = new JBus<>(Date.class, Executors.newCachedThreadPool());
		jbus.register(d -> {
			JBus.deregister();
			System.out.println("the date is:" + d);
		});
		jbus.post(new Date());
		jbus.post(new Date());
		jbus.post(new Date());
```

The above work well for us, so I figured I'd share. Cheers.

**Comment by anidotnet** [comment]
Good work. If you want to create a PR that would be great.
